src/main/java/org/opensearch/jobscheduler/rest/action/RestGetJobDetailsAction.java

```diff
@@ -61,39 +63,30 @@ public String getName() {
     public List<Route> routes() {
         return ImmutableList.of(
             // New Job Details Entry Request
-            new Route(PUT, String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_get/_job_details")),
+            new Route(PUT, String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_job_details")),
             // Update Job Details Entry Request
             new Route(
                 PUT,
-                String.format(
-                    Locale.ROOT,
-                    "%s/%s/{%s}",
-                    JobSchedulerPlugin.JS_BASE_URI,
-                    "_get/_job_details",
-                    GetJobDetailsRequest.DOCUMENT_ID
-                )
+                String.format(Locale.ROOT, "%s/%s/{%s}", JobSchedulerPlugin.JS_BASE_URI, "_job_details", GetJobDetailsRequest.DOCUMENT_ID)
             )
 
         );
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+    public RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
         XContentParser parser = restRequest.contentParser();
         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
 
         GetJobDetailsRequest getJobDetailsRequest = GetJobDetailsRequest.parse(parser);
-
-        final String[] jobDetailsResponseHolder = new String[1];
```

Comment:
I'm guessing this is leftover from a "need a mutable string" Countdown Latch implementation?

Reply:
No actually as we use completetable future instead of countdown latch now, so when we are setting a response of the async call in the local variable then It says to create a final immutable array and then do it. Instead the ideal way is to create a global variable in the class and achieve it. So that's what I did here

Reply:
yeah, I was just saying I think we had the array in there earlier, then it stayed an array when we moved to completable futures.  This change is good.